src/lib.rs

```diff
@@ -430,8 +479,13 @@ impl<K: Hash + Eq, V, S: HashState> LinkedHashMap<K, V, S> {
     /// assert_eq!(None, iter.next());
     /// ```
     pub fn iter(&self) -> Iter<K, V> {
+        let head = if self.head != ptr::null_mut() {
+            unsafe { (*self.head).prev }
+        } else {
+            ptr::null_mut()
```

Comment:
I'd *really* appreciate a comment why this is legit. Presumably there's a guard for head = tail.